dashboard/app/models/user.rb

```diff
@@ -575,6 +575,13 @@ def reset_secret_words
     save!
   end
 
+  def all_advertised_scripts_completed?
```

Comment:
this could be stored as an array in the Script model so it's not duplicated in multiple places?